src/ast.ls

```diff
@@ -1660,10 +1660,14 @@ class exports.Assign extends Node
                     continue if skip and i+2 is len
                     start = i+1
                     @temps = [ivar = o.scope.temporary \i]
-                    val = if skip then node = Var ivar; Var val else
-                        Arr.wrap JS "#i < (#ivar = #val)
-                                  \ ? #{ util \slice }.call(#rite, #i, #ivar)
-                                  \ : (#ivar = #i, [])"
+                    val = switch
+                    | skip
```

Comment:
This patch is separated from summivox/LiveScript@e6591cfe35 . No particular reason.
Change it back to if?